HPS/Heartland/Model/Payment.php

```diff
@@ -239,12 +240,12 @@ public function capture(\Magento\Payment\Model\InfoInterface $payment, $amount)
 
             $this->log($response,'setStatus ');
             /** @var \Magento\Sales\Model\Order\Payment $payment */
-            $payment->setStatus($response->responseText)
-                ->setTransactionId($response->transactionId)
-                ->setIsTransactionClosed(false)
-                ->setCcLast4($CcL4)
-                ->setAdditionalInformation($response->authorizationCode)
-                ->setAmount($amount);
+            @$payment->setStatus($response->responseText);
```

Comment:
Probably works fine the other way but i noticed it behaved unexpectedly with invalid keys